sw/device/sca/aes_serial.c

```diff
@@ -239,6 +276,30 @@ static void aes_serial_single_encrypt(const uint8_t *plaintext,
   aes_send_ciphertext(false);
 }
 
+/**
+ * Advances data for fvsr-key TVLA - fixed set
+ *
+ * This function updates plaintext_fixed for fvsr TVLA, according
+ * to DTR recommendations.
+ */
+static void aes_serial_advance_fixed(void) {
+  aes_encrypt_block(plaintext_fixed, key_gen_round_keys, ciphertext_temp);
+  memcpy(plaintext_fixed, ciphertext_temp, kAesTextLength);
+}
+
+/**
+ * Advances data for fvsr-key TVLA - random set
+ *
+ * This function updates plaintext_random and key_random for fvsr TVLA,
+ * according to DTR recommendations.
+ */
+static void aes_serial_advance_random(void) {
+  aes_encrypt_block(plaintext_random, key_gen_round_keys, ciphertext_temp);
+  memcpy(plaintext_random, ciphertext_temp, kAesTextLength);
+  aes_encrypt_block(key_random, key_gen_round_keys, ciphertext_temp);
```

Comment:
It's a bit unfortunate that we now have `aes_encrypt()` (using the hardware engine) and `aes_encrypt_block()` (using the SW routine for PRNG purposes only). But  I don't know an easy way around this. Maybe rename `aes_encrypt_block()` to `aes_sw_enrcypt()`? What do you think @johannheyszl , @nasahlpa ?

Reply:
For the time being, I've changed this to `aes_sw_encrypt_block()`

Reply:
Sounds good to me, thanks for making the change.